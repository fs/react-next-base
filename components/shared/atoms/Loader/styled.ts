import styled, { css } from 'styled-components';
import { transparentize } from 'polished';

export const Wrapper = styled.div(
  ({ theme: { colors } }) => css`
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background-color: ${transparentize(0.1, colors.white)};
    opacity: 0.7;
  `,
);

export const Title = styled.span`
  font-size: 1.5rem;
  font-weight: bold;
`;
