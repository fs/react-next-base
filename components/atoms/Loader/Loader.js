import React from 'react';
import styled, { css } from 'styled-components';
import { transparentize } from 'polished';

const Wrapper = styled.div(
  ({ theme: { colors } }) => css`
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background-color: ${transparentize(0.1, colors.white)};
    opacity: 0.7;
  `,
);

const Title = styled.span`
  font-size: 1.5rem;
  font-weight: bold;
`;

const Loader = ({ children, testId }) => (
  <Wrapper data-testid={testId} data-cy={testId}>
    <Title>{children}</Title>
  </Wrapper>
);

export default Loader;
