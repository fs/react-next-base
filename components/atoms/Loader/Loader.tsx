import React from 'react';
import type { PropsWithChildren } from 'react';
import styled, { css } from 'styled-components';
import { transparentize } from 'polished';

import type ITest from 'types/testType';

const Wrapper = styled.div(
  ({ theme: { colors } }) => css`
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    background-color: ${transparentize(0.1, colors.white)};
    opacity: 0.7;
  `,
);

const Title = styled.span`
  font-size: 1.5rem;
  font-weight: bold;
`;

type Props = ITest;

const Loader = ({ children, testId }: PropsWithChildren<Props>): JSX.Element => (
  <Wrapper data-testid={testId} data-cy={testId}>
    <Title>{children}</Title>
  </Wrapper>
);

export default Loader;
