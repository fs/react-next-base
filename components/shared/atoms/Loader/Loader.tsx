import React from 'react';
import type { PropsWithChildren } from 'react';

import type Props from 'interfaces/testType';

import { Wrapper, Title } from './styled';

const Loader = ({ children, testId }: PropsWithChildren<Props>): JSX.Element => (
  <Wrapper data-testid={testId} data-cy={testId}>
    <Title>{children}</Title>
  </Wrapper>
);

export default Loader;
