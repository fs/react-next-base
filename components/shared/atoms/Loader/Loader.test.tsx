import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';

import renderWithTheme from '__tests__/helpers/renderWithTheme';
import Loader from 'components/shared/atoms/Loader';

describe('Loader', () => {
  test('should render correctly', async () => {
    // Arrange
    const mockContent = "Loader's content";

    // Act
    render(renderWithTheme(<Loader>{mockContent}</Loader>));

    const loader = screen.getByText(mockContent);

    // Assert
    expect(loader).toMatchSnapshot();
  });
});
