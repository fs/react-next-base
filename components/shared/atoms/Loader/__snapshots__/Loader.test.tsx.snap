// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render correctly 1`] = `
.c0 {
  font-size: 1.5rem;
  font-weight: bold;
}

<span
  class="c0"
>
  Loader's content
</span>
`;
